docs/proposals/29-autoscaling-storage-volumes.md

```diff
@@ -0,0 +1,335 @@
+- Observability volumes are selectively marked for action by said controller
+- Initial volume size is reduced
+- Where the storage infrastructure does not support volume resizing, the preexisting, static-size volume sizing strategy is employed
+
+The current initiative is limited to scaling observability components - _Vali_ and _Prometheus_.
+However, the long term vision is, once the PV autoscaling implementation matures enough, to also use it to offer PV
+auto-scaling for shoot (i.e. customer) volumes.
+
+There are no plans to extend the present initiative to shoot Etcd volumes. Instead, a more advanced, Etcd-specific
+storage management functionality is envisioned in `etcd-druid`. For details, see
+[`etcd-druid` feature request issue#481](https://github.com/gardener/etcd-druid/issues/481#issuecomment-1346402013).
+
+### Goals
+- General ability to autoscale seed PVs.
+  - Ability is conditional on PVC's storage class supporting resizing.
+  - Autoscaling is inactive by default.
+  - Ability to selectively activate autoscaling per PVC.
+  - Ability to trigger scaling based on relative volume utilisation ratio (`<used bytes> / <capacity bytes>`). Trigger
+    threshold configurable via code.
+  - Ability to configure absolute upper scaling limit per PVC.
+  - Autoscaling causes minimal interruption to scaled workloads.
+- Autoscaling ability activated for observability volumes (`prometheus` and `vali`) in seeds'
+  `garden` namespace, and in shoot namespaces.
+- Absolute upper scaling limit configured on each observability volume.
+- Reduce the initial size of a volume, when autoscaling is activated for that volume, if such reduction is aligned with
+  general Gardener goals (cost, reliability).
+- Some metrics, which represent an overview of the PVC autoscaling state per seed, available through one or more of the seed
+  system's Prometheus instances.
+- Provide operational documentation for Gardener Operators.
+- Ability to enable/disable the feature per seed, via feature gate.
+
+### Non-Goals
+- Ability to activate autoscaling on a group of PVCs, other than specifying each PVC individually. Activate by selector
+  or namespace is not supported.
+- Ability to selectively deactivate autoscaling, e.g. by PVC selector or namespace.
+- Ability to scale a PVC if the storage class does not support resizing.
+- Ability to deactivate autoscaling for any of the observability volumes specified in the 'Goals' section.
+- Ability to scale volumes in shoot clusters.
+
+### Notes:
+- Shrinking a PVC is not supported. A limitation imposed by the underlying infrastructure, PVC size can only be
+  increased, and not decreased.
+- The newly introduced auto-scaling mode is predicated on the underlying infrastructure supporting volume resizing.
+  This ability is expressed through the K8s StorageClass type. The respective StorageClass must have
+  `allowVolumeExpansion: true`. For details, see the K8s volume expansion documentation.
```

Comment:
Posting this here for reference.

* https://kubernetes.io/docs/concepts/storage/storage-classes/#allow-volume-expansion
* https://kubernetes.io/blog/2018/07/12/resizing-persistent-volumes-using-kubernetes/

Otherwise, I agree - please add these links to the GEP.

Reply:
Yes, please add those links to the GEP.

Reply:
Done. Will be available with next push.
@dnaeon, thanks for the links!